File: backend/src/main/java/com/campussync/campussync_backend/entity/Student.java
package com.campussync.campussync_backend.entity;

import com.campussync.campussync_backend.enums.StudentStatus;

import jakarta.persistence.*;

import jakarta.validation.constraints.NotBlank;
import java.time.LocalDate;
import jakarta.validation.constraints.NotNull;

import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "students")
@Getter
@Setter
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @NotBlank
    private String registerNumber;

    @ManyToOne
    @JoinColumn(name = "department_id", nullable = false)
    private Department department;

    @ManyToOne
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @NotNull
    private Integer semester;

    private boolean internal;

    private String collegeName;
    @NotBlank
private String programme;

@NotNull
private Integer admissionYear;

    private Integer graduationYear;


    @NotNull
    @Enumerated(EnumType.STRING)
    private StudentStatus status;
    private LocalDate dateOfBirth;
}

File: frontend/lib/models/student.dart
class Student {
  final int id;
  final String name;
  final String email;
  final String registerNumber;

  final int departmentId;
  final String departmentName;

  final int branchId;
  final String branchName;

  final String programme;
  final int admissionYear;
  final int semester;
  final int graduationYear;

  final bool internal;
  final String status;
  final String accountStatus;

  Student({
    required this.id,
    required this.name,
    required this.email,
    required this.registerNumber,
    required this.departmentId,
    required this.departmentName,
    required this.branchId,
    required this.branchName,
    required this.programme,
    required this.admissionYear,
    required this.semester,
    required this.graduationYear,
    required this.internal,
    required this.status,
    required this.accountStatus,
  });

  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
      id: json['id'],
      name: json['name'],
      email: json['email'],
      registerNumber: json['registerNumber'],
      departmentId: json['departmentId'],
      departmentName: json['departmentName'],
      branchId: json['branchId'],
      branchName: json['branchName'],
      programme: json['programme'],
      admissionYear: json['admissionYear'],
      semester: json['semester'],
      graduationYear: json['graduationYear'],
      internal: json['internal'] ?? false,
      status: json['status'] ?? '',
      accountStatus: json['accountStatus'] ?? '',
    );
  }
} 